src/main/java/io/mosip/digitalcard/service/impl/PDFCardServiceImpl.java

```diff
@@ -0,0 +1,568 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.net.URI;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.*;
+
+@Service
+public class PDFCardServiceImpl implements PDFCardService {
+
+	@Value("${mosip.digitalcard.websub.publish.topic:CREDENTIAL_STATUS_UPDATE}")
+	private String topic;
+	
+	@Autowired
+	private WebSubSubscriptionHelper webSubSubscriptionHelper;
+
+	@Autowired
+	private DataShareUtil dataShareUtil;
+
+	@Autowired
+	private RestClient restApiClient;
+
+	/** The Constant FILE_SEPARATOR. */
+	public static final String FILE_SEPARATOR = File.separator;
+
+	/** The Constant VALUE. */
+	private static final String VALUE = "value";
+
+	@Value("${mosip.digitalcard.templateTypeCode:RPR_UIN_CARD_TEMPLATE}")
+	private String uinCardTemplate;
+
+	/** The Constant FACE. */
+	private static final String FACE = "Face";
+
+	/** The Constant UIN_TEXT_FILE. */
+	private static final String UIN_TEXT_FILE = "textFile";
+
+	/** The Constant APPLICANT_PHOTO. */
+	private static final String APPLICANT_PHOTO = "ApplicantPhoto";
+
+	/** The Constant QRCODE. */
+	private static final String QRCODE = "QrCode";
+
+	/** The Constant UINCARDPASSWORD. */
+	private static final String UINCARDPASSWORD = "mosip.digitalcard.uincard.password";
+
+	/** The print logger. */
+	Logger printLogger = DigitalCardRepoLogger.getLogger(PDFCardServiceImpl.class);
+
+	/** The template generator. */
+	@Autowired
+	private TemplateGenerator templateGenerator;
+
+	/** The utilities. */
+	@Autowired
+	private Utility utility;
+
+	@Autowired
+	private EncryptionUtil encryptionUtil;
+
+	/** The uin card generator. */
+	@Autowired
+	private UinCardGenerator<byte[]> uinCardGenerator;
+
+	/** The qr code generator. */
+	@Autowired
+	private QrCodeGenerator<QrVersion> qrCodeGenerator;
+
+	/** The cbeffutil. */
+	@Autowired
+	private CbeffUtil cbeffutil;
+
+	/** The env. */
+	@Autowired
+	private Environment env;
+
+	@Autowired
+	DigitalCardTransactionRepository digitalCardTransactionRepository;
+
+	@Autowired
+	private CredentialsVerifier credentialsVerifier;
+
+	@Value("${mosip.digitalcard.datashare.partner.id}")
+	private String partnerId;
+
+	@Value("${mosip.digitalcard.datashare.policy.id}")
+	private String policyId;
+
+	@Value("${mosip.template-language}")
+	private String templateLang;
+
+	@Value("#{'${mosip.mandatory-languages:}'.concat('${mosip.optional-languages:}')}")
+	private String supportedLang;
+
+	@Value("${mosip.digitalcard.verify.credentials.flag:true}")
+	private boolean verifyCredentialsFlag;
+
+	@Value("${mosip.digitalcard.pdf.password.enable.flag:true}")
+	private boolean pdfPasswordFlag;
+
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	public PDFCardServiceImpl() {
+	}
+
+
+	public boolean generateCard(EventModel eventModel) {
```

Comment:
I see this layer is doing the download of VC, verifying it and then using it to generate the pdf template. Can we move the download and verification of VC to the Digitial card service layer ? So this layer does not need to understand the event model instread it taken in a VC and then generates a PDF out of that.
Also later if we decide to support other formats, the interface will be clean and we can avoid duplicate code.

Reply:
taken care

Reply:
Please keep method name as generateCard, so we can give mutiple implementation of the cards

Reply:
the name is taken care

Reply:
As discussed, please remove unwanted arguments like request id and pass the password directly instead of boolean (Password generation logic should move to digitial card service). 
Also lets make sure that password values are taken from received VC, we should not call idrepo API to get any details

Reply:
taken care